src/shogun/mathematics/linalg/GPUMemoryViennaCL.h

```diff
@@ -42,14 +42,21 @@ struct GPUMemoryViennaCL : public GPUMemoryBase<T>
 		viennacl::backend::memory_write(*m_data, 0, m_len*sizeof(T), vector.vector);
 	}
 
+	GPUMemoryBase<T>* clone_vector(const SGVector<T>& vector) const
```

Comment:
this interface is weird.
It should take a GPUMemoryBase and just create a newly allocated copy of the data.
Then, when you call it from SGVector, you should just pass the GPUMemoryBase pointer and length